sky/engine/core/dart/painting.dart

```diff
@@ -0,0 +1,12 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+part of dart_ui;
+
+typedef void _ImageDecoderCallback(Image result);
+
+class PaintingNatives {
+  static void decodeFromDataPipe(int handle, _ImageDecoderCallback callback)
+      native "PaintingNatives_decodeFromDataPipe";
+}
```

Comment:
There no need for this PaintingNatives class.  Instead, we can just have a top-level function called `decodeImageFromDataPipe`